### PR TITLE
drogon: Add option to use spdlog

### PR DESCRIPTION
**Pull request by matthiasbuhl**
Library name and version:  **drogon/1.9.2**

As of **version 1.9.2**, drogon integrated the feature to use spdlog as the logging backend. 
The changes of this PR add support to the package to support this new feature.

---

- [X] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [X] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [X] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by matthiasbuhl** [comment]
The build seems to have failed but I cannot view any output of the build due to access restriction.
Can someone let me know what seems to be failing or how I can reproduce this locally?

**Comment by matthiasbuhl** [comment]
@ericLemanissier and @prince-chrismc, 
did you have a chance to look at the changes made? Could you let me know why it might not be building according to the Github checks?

**Comment by ericLemanissier** [comment]
Please read carefully all the messages above (in this PR)

**Review by uilianries — CHANGES_REQUESTED** [review]
The spdlog is a new feature, only available since 1.9.2. It should not be available as option to older versions. 

https://github.com/drogonframework/drogon/commit/8026790e1a8e1f3baf37bd2ecb822a1872933c55

Something like:

```
def config_options(self):
    if Version(self.version) < "1.9.2":
        del self.options.with_spdlog
```

**Comment by matthiasbuhl** [comment]
> Thello @matthiasbuhl !! Thank you for your first contribution!
> 
> The spdlog is a new feature, only available since 1.9.2. It should not be available as option to older versions.
> 
> [drogonframework/drogon@8026790](https://github.com/drogonframework/drogon/commit/8026790e1a8e1f3baf37bd2ecb822a1872933c55)
> 
> Something like:
> 
> ```
> def config_options(self):
>     if Version(self.version) < "1.9.2":
>         del self.options.with_spdlog
> ```

Thanks for the feedback, Ill get right on it.

**Comment by matthiasbuhl** [comment]
> Thello @matthiasbuhl !! Thank you for your first contribution!
> 
> The spdlog is a new feature, only available since 1.9.2. It should not be available as option to older versions.
> 
> [drogonframework/drogon@8026790](https://github.com/drogonframework/drogon/commit/8026790e1a8e1f3baf37bd2ecb822a1872933c55)
> 
> Something like:
> 
> ```
> def config_options(self):
>     if Version(self.version) < "1.9.2":
>         del self.options.with_spdlog
> ```

Should be fixed!

File: recipes/drogon/all/conanfile.py [hunk 1]
```diff
@@ -70,6 +74,7 @@ def configure(self):
             del self.options.with_mysql
             del self.options.with_sqlite
             del self.options.with_redis
+            del self.options.with_spdlog
```
Comment (uilianries) [inline recipes/drogon/all/conanfile.py:77]:
```suggestion
           self.options.rm_safe("with_spdlog")
```

double deletion is an error in Conan 2.x.

File: recipes/drogon/all/conanfile.py [hunk 2]
```diff
@@ -154,6 +161,7 @@ def generate(self):
         tc.variables["BUILD_MYSQL"] = self.options.get_safe("with_mysql", False)
         tc.variables["BUILD_SQLITE"] = self.options.get_safe("with_sqlite", False)
         tc.variables["BUILD_REDIS"] = self.options.get_safe("with_redis", False)
+        tc.variables["USE_SPDLOG"] = self.options.get_safe("with_spdlog", False)
```
Comment (uilianries) [inline recipes/drogon/all/conanfile.py:164]:
```suggestion
        if Version(self.version) > "1.9.1":
            tc.variables["USE_SPDLOG"] = self.options.get_safe("with_spdlog", False)
```
So we avoid cmake warnings to older versions.

**Review by uilianries — COMMENTED** [review]
Looking good, only small details for safety.

**Comment by matthiasbuhl** [comment]
Alright I think it should all be in now. Thanks for the help!

**Comment by uilianries** [comment]
In terms of recipe it looks really good. Just waiting next authorized users wave to include you, then the CI will build this PR. Regards!

**Comment by matthiasbuhl** [comment]
> I detected other pull requests that are modifying drogon/all recipe:
> 
> * [drogon: add version 1.9.3 #22730](https://github.com/conan-io/conan-center-index/pull/22730)
> 
> This message is automatically generated by https://github.com/ericLemanissier/conan-center-conflicting-prs so don't hesitate to report issues/improvements there.

Just checked this PR but it is not really conflicting as this PR only updates to the most recent drogon release but still does not bring in the option to activate the spdlog support that drogon introduced with 1.9.2

**Pull request closed without merging**